lib/we/call/middleware/client/detect_deprecations.rb

```diff
@@ -42,13 +42,21 @@ def report_deprecated_usage(env, datetime)
             else
               warning = "Endpoint #{env.url} was deprecated for removal on #{datetime.iso8601} and could be removed AT ANY TIME"
             end
+            send_warning warning
+          end
 
+          def send_warning(warning)
+            warned = false
             if @options[:active_support]
               ActiveSupport::Deprecation.warn(warning)
-            elsif @options[:logger] && @options[:logger].respond_to?(:warn)
+              warned = true
+            end
+            if @options[:logger] && @options[:logger].respond_to?(:warn)
               @options[:logger].warn(warning)
-            else
-              raise NoOutputForWarning, "Pass active_support: true or logger: SomeLoggerWithWriteMethod when registering middleware"
+              warned = true
+            end
+            unless warned
+              raise NoOutputForWarning, "Pass active_support: true, or logger: ::Logger.new when registering middleware"
```

Comment:
wouldn't this be an output we could detect at config load?

Reply:
Followup commit has actually moved this file to its own faraday-sunset gem, so this won't be a thing anymore.